server/app/assets/javascripts/sort_selector.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * Responsible for controlling the sorting mechanism used by QuestionBankController. Elements (tagged with elementSelector class)
+ * have data attributes with the value to be compared and will sort list using the value. The data attribute is defined
+ * in QuestionSortOption.
+ *
+ * E.g. Sort by Name A-Z will look up the value for data attribute tiname-asc and sort the list by that value.
+ *
+ * @param listId id for finding the portion of the document that contains the sublistSelector and elementSelector
+ * @param sublistSelector selector for the parent element of all of the elements to be sorted
```

Comment:
(non-blocking) the sublistSelector can return multiple entities, and this method happily sorts each of them separately, which is not clear from this line